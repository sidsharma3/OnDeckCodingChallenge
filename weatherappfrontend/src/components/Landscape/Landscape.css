* {
    box-sizing: border-box;
    padding: 0;
    margin: 0;
}

.landscape {
    height: 100vh;
    width: 100%;
    background-color: skyblue;
    display: flex;
    flex-direction: row;
    align-items: center;
    text-align: center;
    justify-content: center;
}

.land {
    z-index: 2;
    position: absolute;
    top: 100%;
    left: 50%;
    transform: translate(-50%, -50%);
    height: 30%;
    width: 100%;
    border-radius: 30% 30% 0 0;
    background-color: green;
    border: 3px solid;
}

.sun {
    z-index: 1;
    position: absolute;
    top: 0px;
    right: 0px;
    height: 250px;
    width: 250px;
    line-height: 75px;
    text-align: center;
    background: orange;
    border-bottom-left-radius: 100%;
    border: 3px solid;
}
    
