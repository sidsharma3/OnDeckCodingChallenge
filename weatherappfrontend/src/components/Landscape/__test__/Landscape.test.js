import React from 'react'
import ReactDOM from 'react-dom'
import Landscape from '../Landscape'
import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect"
import renderer from "react-test-renderer"

afterEach(cleanup)
it("Landscape component rendered without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Landscape/>, div)
})

it("Matches Landscape component Snapshot", () => {
    const landscape = renderer.create(<Landscape/>).toJSON()
    expect(landscape).toMatchSnapshot();
})