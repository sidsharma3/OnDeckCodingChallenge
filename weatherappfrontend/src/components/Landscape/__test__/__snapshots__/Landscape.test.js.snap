// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Landscape component Snapshot 1`] = `
<div
  className="landscape"
>
  <div
    className="weatherDataBox"
  >
    <div
      className="cityInputBox"
    >
      <input
        onKeyDown={[Function]}
        placeholder="Type a City Here and Press Enter to get the Weather!"
        type="text"
      />
    </div>
    <h1
      data-testid="citynotFoundText"
    >
       Could Not Find Weather Data on City 
    </h1>
  </div>
  <div
    className="land"
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  />
  <div
    className="sun"
  />
</div>
`;
